desktop-widgets/tab-widgets/TabDivePhotos.cpp

```diff
@@ -49,32 +49,43 @@ void TabDivePhotos::clear()
 void TabDivePhotos::contextMenuEvent(QContextMenuEvent *event)
 {
 	QMenu popup(this);
-	popup.addAction(tr("Load image(s) from file(s)"), this, SLOT(addPhotosFromFile()));
-	popup.addAction(tr("Load image(s) from web"), this, SLOT(addPhotosFromURL()));
+	popup.addAction(tr("Load image(s) from file(s)"), this, &TabDivePhotos::addPhotosFromFile);
+	popup.addAction(tr("Load image(s) from web"), this, &TabDivePhotos::addPhotosFromURL);
 	popup.addSeparator();
-	popup.addAction(tr("Delete selected images"), this, SLOT(removeSelectedPhotos()));
-	popup.addAction(tr("Delete all images"), this, SLOT(removeAllPhotos()));
+	popup.addAction(tr("Delete selected images"), this, &TabDivePhotos::removeSelectedPhotos);
+	popup.addAction(tr("Delete all images"), this, &TabDivePhotos::removeAllPhotos);
+	popup.addAction(tr("Recalculate selected thumbnails"), this, &TabDivePhotos::recalculateSelectedThumbnails);
 	popup.exec(event->globalPos());
 	event->accept();
 }
 
-void TabDivePhotos::removeSelectedPhotos()
+QVector<QString> TabDivePhotos::getSelectedFilenames() const
 {
 	if (!ui->photosView->selectionModel()->hasSelection())
-		return;
-	QModelIndexList indexes =  ui->photosView->selectionModel()->selectedRows();
+		return {};
```

Comment:
Ok - did that, since it is probably a more well-known idiom.

But as far as I know, `return {};` is the same as `return QVector();`. With mandatory copy-elision it will not construct an object on the stack, but directly construct an empty-vector in the memory provided by the caller. So it should be very efficient.